File: peek_plugin_docdb/plugin-module/_private/tuples/DocDbTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {docDbTuplePrefix} from "../PluginNames";


@addTupleType
export class DocDbTuple extends Tuple {
    public static readonly tupleName = docDbTuplePrefix + "DocDbTuple";

    //  Description of date1
    dict1 : {};

    //  Description of array1
    array1 : any[];

    //  Description of date1
    date1 : Date;

    constructor() {
        super(DocDbTuple.tupleName)
    }
}

File: peek_plugin_docdb/_private/tuples/DocDbTuple.py
from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_docdb._private.PluginNames import docDbTuplePrefix


@addTupleType
class DocDbTuple(Tuple):
    """ DocDb Tuple

    This tuple is a create example of defining classes to work with our data.
    """
    __tupleType__ = docDbTuplePrefix + 'DocDbTuple'

    #:  Description of date1
    dict1 = TupleField(defaultValue=dict)

    #:  Description of date1
    array1 = TupleField(defaultValue=list)

    #:  Description of date1
    date1 = TupleField()
